src/libraries/Common/tests/System/Net/Sockets/TestPortPool.cs

```diff
@@ -0,0 +1,230 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.Security;
+using System.Net.Test.Common;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.DotNet.RemoteExecutor;
+
+namespace System.Net.Sockets.Tests
+{
+    internal readonly struct PortLease : IDisposable
```

Comment:
Instead of managing the ports with `PortLease`, could we incrementally try ports from a range. If the range is large enough for all tests that need a unique port nr, we don't need to deal with returning (and possible issues by re-using port nrs).
There can be `BindToTcpPoolPort`/`BindToUdpPoolPort` methods which are similar to `BindToAnonymousPort`. (TCP and UDP ports are distinct).

Reply:
That would mean creating and destroying temporary sockets, since we need to probe both IPV4 and IPV6.

I was afraid of possible side effects which are unknown to me, and rent/return seemed less risky. If you say there are none, I believe you :)

Reply:
> That would mean creating and destroying temporary sockets, since we need to probe both IPV4 and IPV6.

Why is probing needed?
If we incrementally use ports from the range, and not re-use any, do we still expect issues?

Reply:
>  could we incrementally try ports from a range

I thought you meant probing by "trying" here. It seemed logical to probe both protocols to make sure we are safe. 

But you are right, we don't need probing *at all*, if the port range is big enough compared to the number of tests. In that case I would just return the next port number, and let the test fail in the theoretical case of a collision (which we don't expect to happen in practice.)

I will remove the returning logic, and it's backing dictionary. Seems like a reasonable simplification, thanks for the suggestion!